composer/cli/launcher.py

```diff
@@ -92,17 +104,18 @@ def launch_processes(nproc: int, world_size: int, base_rank: int, master_addr: s
         current_env["LOCAL_WORLD_SIZE"] = str(nproc)
         current_env["MASTER_ADDR"] = master_addr
         current_env["MASTER_PORT"] = str(master_port)
+        current_env["RUN_DIRECTORY"] = run_directory
 
-        print(f"Launching process for local_rank({local_rank}), global_rank({global_rank})", local_rank, global_rank)
+        print(f"Launching process for local_rank({local_rank}), global_rank({global_rank})")
 
         if local_rank == 0:
             process = subprocess.Popen(cmd, env=current_env, text=True)
```

Comment:
Is there a way to have the rank 0 process print to `stdout` _and_ a log file?

Reply:
Theoretically yes, though we would have to create our own file writer. My concern about having our own one would be re: buffering and TQDM, which is why I skipped it. But I agree this would be really nice to have.

Reply:
(Experimenting with it more, python subprocess require a real FD. So, we'd have to write to a file, and then have a background thread to read from stdout/stderr, and reinterpret TQDM and terminal-like commands. This won't be trivial...)

Reply:
Ok, that doesn't sound like a good use of our time :)